components/molecules/m-product-gallery.vue

```diff
@@ -57,11 +55,19 @@ export default {
       };
     },
     variantImage () {
-      let variantImage = this.gallery.find(
-        imageObject =>
-          isEqual(imageObject.id, this.option) ||
-          (imageObject.id && imageObject.id.color === this.option.color)
-      );
+      let variantImage = this.gallery.find(image => {
+        let selectThis = true
+        for (const [key, value] of Object.entries(this.configuration)) {
+          if (
+            typeof image.id !== 'undefined' &&
+            typeof image.id[key] !== 'undefined' &&
+            image.id[key] !== value.id
+          ) {
+            selectThis = false
+          }
+        }
+        return selectThis || (image.id && image.id.color === this.configuration.color.id)
```

Comment:
Now it will return first variant image that matches only the color, regardless of any other images - even if they fit better (because they may contain more matching properties to current product configuration).

If it should work like that then it's fine, but in that case the whole logic for `variantImage` could be simplified: there is no need to iterate over all properties in current configuration.

Reply:
Fixed